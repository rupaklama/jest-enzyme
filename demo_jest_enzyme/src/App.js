import Congrats from './Congrats';
import GuessedWords from './GuessedWords';
import Input from './Input';

function App() {
  // TODO: data will come through context api
  const success = false;
  const secretWord = 'party';
  const guessedWords = [];

  return (
    <main data-test='component-app'>
      <h1>Jotto - Guess the Word ?</h1>
      <Input secretWord={secretWord} />
      <Congrats success={success} />
      <GuessedWords guessedWords={guessedWords} />
    </main>
  );
}

export default App;
