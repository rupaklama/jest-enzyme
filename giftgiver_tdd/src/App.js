const App = () => {
  return (
    <main>
      <h2>Snapshot Testing</h2>
    </main>
  );
};

export default App;
