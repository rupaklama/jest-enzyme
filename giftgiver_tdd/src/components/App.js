import React, { useState } from 'react';

const App = () => {
  const [gifts, setGifts] = useState([]);

  return (
    <div>
      <h2>Gift Giver</h2>
    </div>
  );
};

export default App;
