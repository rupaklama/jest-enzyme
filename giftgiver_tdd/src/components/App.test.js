import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { toMatchDiffSnapshot } from 'snapshot-diff';

import App from './App';

expect.extend({ toMatchDiffSnapshot });

// default state
const setup = (state = {}) => {
  return shallow(<App />);
};

describe('rendering', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup({
      gifts: [],
    });
  });

  test('should render component correctly', () => {
    const component = toJson(wrapper);
    expect(component).toMatchSnapshot();

    const componentUpdate = toJson(wrapper);
    expect(componentUpdate).toMatchDiffSnapshot(component);
  });

  test('initializes the `state` with an empty list of gifts', () => {});
});
