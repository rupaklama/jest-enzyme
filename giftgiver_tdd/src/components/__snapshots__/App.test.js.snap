// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render component correctly 1`] = `
<div>
  <h2>
    Gift Giver
  </h2>
</div>
`;

exports[`rendering should render component correctly 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
